crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1622,4 +1707,117 @@ mod tests {
+            // in the roomInfo
+            origin_server_ts: timestamp(0),
+            state_key: user,
+            unsigned: StateUnsigned::new(),
+        }))
+    }
+
+    fn membership_for_my_call(
+        device_id: &str,
+        membership_id: &str,
+        minutes_ago: u32,
+    ) -> Membership {
+        let application = Application::Call(CallApplicationContent::new(
+            "my_call_id_1".to_owned(),
+            ruma::events::call::member::CallScope::Room,
+        ));
+        let foci_active = vec![Focus::Livekit(LivekitFocus::new(
+            "my_call_foci_alias".to_owned(),
+            "https://lk.org".to_owned(),
+        ))];
+
+        assign!(
+            Membership::from(MembershipInit {
+                application,
+                device_id: device_id.to_owned(),
+                expires: Duration::from_millis(3_600_000),
+                foci_active,
+                membership_id: membership_id.to_owned(),
+            }),
+            { created_ts: Some(timestamp(minutes_ago)) }
+        )
+    }
+
+    fn receive_state_events(room: &Room, events: Vec<&AnySyncStateEvent>) {
+        let mut guard = room.inner.write();
+        ObservableWriteGuard::update(&mut guard, |g| {
+            for e in events {
+                g.handle_state_event(e);
+            }
+        });
+    }
+
+    /// `user_a`: empty memberships
+    /// `user_b`: one membership
+    /// `user_c`: two memberships (two devices)
+    fn create_call_with_member_events_for_user(user_a: &str, user_b: &str, user_c: &str) -> Room {
+        let (_store, room) = make_room(RoomState::Joined);
+
+        // make b 10min old
+        let m_init_b = membership_for_my_call("0", "0", 10);
+        // c1 1min old
+        let m_init_c1 = membership_for_my_call("0", "0", 1);
+        // c2 20min old
+        let m_init_c2 = membership_for_my_call("1", "0", 20);
+
+        let a_empty = call_member_state_event(Vec::new(), "$1234", user_a);
+        let b_one = call_member_state_event(vec![m_init_b], "$12345", user_b);
+        let c_two = call_member_state_event(vec![m_init_c1, m_init_c2], "$123456", user_c);
+
+        receive_state_events(&room, vec![&a_empty, &b_one, &c_two]);
+
+        room
+    }
+
+    #[test]
+    fn show_correct_active_call_state() {
+        let (user_a, user_b, user_c) = ("@userA:abc.abc", "@userB:abc.abc", "@userC:abc.abc");
+
+        let room = create_call_with_member_events_for_user(user_a, user_b, user_c);
+
+        // This check also test the ordering.
+        // we want older events to be in the front.
+        // b is 10min old, c1 1min old, c2 20min old
+        assert_eq!(vec![user_c.clone(), user_b, user_c], room.active_room_call_participants());
+        assert!(room.has_active_room_call());
+    }
+
+    #[test]
+    fn active_call_is_false_when_everyone_left() {
+        let (user_a, user_b, user_c) = ("@userA:abc.abc", "@userB:abc.abc", "@userC:abc.abc");
```

Comment:
`matrix_sdk_test` has constants `ALICE`, `BOB` and `CAROL`, probably easier to use those instead of declaring user variables in every test. Then you can also update fn signatures to take `&UserId` instead of `&str` and don't need as much `unwrap()`ping inside.

Reply:
That makes things much nicer. 
Is there also sth pretty that I can do with the event_id (some constants or a fn that automatically genenerates an id?)

Reply:
Not sure, maybe keep those as local bindings for now and we'll revisit later.